tracker_linux.go

```diff
@@ -140,13 +140,14 @@ func (t *linuxProcessTracker) doCleanupByEnvironmentVariable() error {
 // cleanupProcess terminates the process given by its PID. If the process has
 // already terminated, this will not return an error.
 func cleanupProcess(pid int) error {
-	catcher := grip.NewBasicCatcher()
 	// A process returns syscall.ESRCH if it already terminated.
-	if err := syscall.Kill(pid, syscall.SIGTERM); err != syscall.ESRCH {
-		catcher.Add(errors.Wrapf(err, "failed to send sigterm to process with pid '%d'", pid))
-		catcher.Add(errors.Wrapf(syscall.Kill(pid, syscall.SIGKILL), "failed to send sigkill to process with pid '%d'", pid))
+	if err := syscall.Kill(pid, syscall.SIGTERM); err != nil && err != syscall.ESRCH {
+		catcher := grip.NewBasicCatcher()
+		catcher.Add(errors.Wrapf(err, "sending sigterm to process with PID '%d'", pid))
+		catcher.Add(errors.Wrapf(syscall.Kill(pid, syscall.SIGKILL), "sending sigkill to process with PID '%d'", pid))
```

Comment:
should we give the process time to cleanup before we follow up with SIGKILL?

Reply:
No, I don't think it's worth waiting. Cleanup means kill everything as soon as possible.

Reply:
it there a reason not to SIGKILL as a first resort, then?

Reply:
Because SIGKILL won't clean up resources, but SIGTERM can be ignored. https://major.io/2010/03/18/sigterm-vs-sigkill/

Reply:
That's what I was getting at in the beginning, though: if we send the two signals in quick succession the process won't have a chance to clean up before it gets terminated.
As an aside, I found [this comic](https://turnoff.us/geek/dont-sigkill/) simultaneously humorous and terrible.

Reply:
Then I guess this begs the question - if we should just trust that processes will obey SIGTERM, we can just send SIGTERM and no error means it's dead. I think that's a reasonable solution, which was the one I originally wrote (i.e. no error from SIGTERM means it's dead).

Reply:
I don't think we can _trust_ processes to obey SIGTERM: the idea is to give them a chance to do cleanup as a curtesy. We can't **know** they'll be dead until we've sent SIGKILL.

Reply:
Sure, but this operation is supposed to be fast. If we have to wait for an unknown amount of time for the process to finish, it'll make this operation expensive to use.